org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/DocumentLifeCycleHandler.java

```diff
@@ -134,10 +134,15 @@ private IStatus performValidation(IProgressMonitor monitor) throws JavaModelExce
 		List<CompilationUnit> astRoots = this.sharedASTProvider.getASTs(toValidate, monitor);
 		for (CompilationUnit astRoot : astRoots) {
 			// report errors, even if there are no problems in the file: The client need to know that they got fixed.
-			DiagnosticsHandler handler = new DiagnosticsHandler(connection, (ICompilationUnit) astRoot.getTypeRoot());
+			ICompilationUnit unit = (ICompilationUnit) astRoot.getTypeRoot();
+			DiagnosticsHandler handler = new DiagnosticsHandler(connection, unit);
 			handler.beginReporting();
-
+			boolean isOnClassPath = JDTUtils.isOnClassPath(unit);
+			boolean isDefaultProject = JDTUtils.isDefaultProject(unit);
 			for (IProblem problem : astRoot.getProblems()) {
+				if (!isDefaultProject && !isOnClassPath && ((problem.getID() & IProblem.Syntax) == 0)) {
```

Comment:
no, error filtering should be done by [DiagnosticsHandler](https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/DiagnosticsHandler.java#L43). So isOnClassPath/isDefaultProject computations could be done in the constructor instead.